### ISSUE TITLE
tests.rest.load_packaged defaults to true in IDEs, loads all default ES REST tests instead of plugin's tests

### ISSUE THREAD

**Issue by dweiss** [elastic/elasticsearch#14109]
Something is wrong with this. A REST IT test case in a project runs against all (or at least a lot) of ES tests, not the project's REST YAMLs.

The examples in ES repository have this:
```
<tests.rest.load_packaged>false</tests.rest.load_packaged>
```
but for any IDE this has to be passed manually. The problem seems to be here, the origin codebase is hardcoded to FileUtils?

https://github.com/elastic/elasticsearch/blob/master/core/src/test/java/org/elasticsearch/test/rest/ESRestTestCase.java#L243

**Comment by rmuir** [comment]
Its just a typical IDE test issue. There is no jenkins job running tests from these IDEs or anything, yet they are essentially alternative build systems. 

The general problem is here: https://github.com/elastic/elasticsearch/issues/13914

One possible solution is to avoid system properties for this kind of thing entirely (instead, generate a resource file, as that can then work everywhere).

**Comment by dweiss** [comment]
I understand the problem in general, but this property is replicated across example plugins as well (it's set to false) and I think the only reason why it's set to false in every one of these plugins is because it wouldn't work otherwise (read: it wouldn't be testing the plugin's REST YAML tests).

**Comment by rmuir** [comment]
Right, but there are bigger issues here: 

* rest tests from the IDE do not work in core or plugins at all right now. I believe this was broken sometime when moving stuff to test-resources
* rest resources are applied in an inconsistent way: you get a test-resource folder named `target/dev-tools/rest-api-spec` which means resources are rooted under `/api` and `/test`
* on the other hand plugin rest resources are situated under `src/test/resources` with resources rooted under `rest-api-spec/api` and `rest-api-spec/test` and so on: totally inconsistent.

I am willing to fix all this up, so that it works from IDEs everywhere intuitively out of box, because I do think its important, but it is a significant amount of work and I don't want to just have to do it again in a few weeks because something like gradle is introduced. Things like IDE support are not tested by jenkins so I give a 100% chance I would have to redo everything again.

**Comment by dweiss** [comment]
Hmm... these tests do work for me out of the box on 2.0.0.rc1. I followed an example from plugins/jvm-* (something, can't remember). 

But I agree with you that it's inconsistent and hard to understand -- I spent a good while trying to figure out what's exactly happening and why (and eventually I gave up on trying to understand it).

This isn't a priority for me anyway as I couldn't figure out the yml "test api" language syntax. Specifically, it didn't seem extensible enough to cover some of the things I wanted to verify (I missed a list-contains-element assertion, for example). Not a big deal, really.
